packages/bot-web-ui/src/stores/main-content-store.js

```diff
@@ -1,11 +1,16 @@
-import { action, observable } from 'mobx';
+import { action, observable, when } from 'mobx';
 import { onWorkspaceResize } from '@deriv/bot-skeleton';
 import { tabs_title } from '../constants/bot-contents';
 import { storeSetting, getSetting } from '../utils/settings';
 
 export default class MainContentStore {
     constructor(root_store) {
         this.root_store = root_store;
+
+        when(
+            () => Blockly.derivWorkspace,
+            () => this.setContainerSize
+        );
```

Comment:
`Blockly.derivWorkspace` is not an observable so this will never trigger. What needs to happen? If this is something that should run once a component is loaded I suggest to use React's `useEffect` instead.

Reply:
@aaron-binary It happens, `when observes and runs the given predicate function until it returns true. Once that happens, the given effect function is executed and the autorunner is disposed.`
https://mobx.js.org/reactions.html
But you think using `UseEffect` is more appropriate here?

Reply:
@aaron-binary The problem is that only after the first login, the `workspace` does not initial in a proper time and we will get an error because it's empty, I only wanted to rerun this function after the workspace initialized.

Reply:
The predicate being something observable, i.e. `It only responds to changes in observable state, things you have annotated observable or computed.`. If that's the case, perhaps we can put the logic in `dbot.js`?

Reply:
@aaron-binary  Don't know what to change there, but since the problem happens in `setContainerSize` which is in `main-content` I know I should rerun it again after the workspace initializes completely.

Reply:
💰: The `initWorkspace` function in `deriv-app/packages/bot-skeleton/src/scratch/dbot.js` returns a promise. In `deriv-app/packages/bot-web-ui/src/stores/app-store.js` on line 21 you can see the caller of this promise, you can put the logic inside the `then` callback there.